webui/react/src/pages/F_ExpList/glide-table/ColumnPickerMenu.tsx

```diff
@@ -33,16 +36,18 @@ interface ColumnMenuProps {
   initialVisibleColumns: string[];
   projectColumns: Loadable<ProjectColumn[]>;
   setVisibleColumns: (newColumns: string[]) => void;
+  projectId: number;
 }
 
 interface ColumnTabProps {
-  columnState: Set<string>;
+  columnState: string[];
```

Comment:
question: Looks like `columnState` is read/write heavy variable. Set seems more efficient. why replacing `Set<string>` with `string[]`?

Reply:
I wanted to preserve the order of columns, to determine if unchecked column is pinned

Reply:
Also I don't think `columnState` would change if we don't make changes to columns?

Reply:
true, it just read only. if its applicable, immutable.js offers [ordered set](https://immutable-js.com/docs/v4.3.0/OrderedSet/)

Reply:
Got it.. I misunderstood